### PR TITLE
Allow 640x480 depth images

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
This PR allows setting the resolution of depth images for the OpenNI driver. Althought it is slow over the network, it might still be desired to have a full resolution depth image in some cases.
Furthermore, the code is cleaned up a little and easier to understand. Also, there are only very few special cases for the Kinect/Xtion differentiation.

Would be great if someone could merge this :).

Thanks,
Tobias

### CODE REVIEW

File: library/src/kinectWrapper_server.cpp [hunk 1]
```diff
@@ -125,11 +127,13 @@ bool KinectWrapperServer::open(const Property &options)
...
     img_width=opt.check("img_width",Value(320)).asInt();
     img_height=opt.check("img_height",Value(240)).asInt();
+    depth_width=opt.check("depth_width",Value(320)).asInt();
+    depth_height=opt.check("depth_height",Value(240)).asInt();
 
-    buf=new unsigned short[KINECT_TAGS_DEPTH_WIDTH*KINECT_TAGS_DEPTH_HEIGHT];
-    bufPl=new unsigned short[KINECT_TAGS_DEPTH_WIDTH*KINECT_TAGS_DEPTH_HEIGHT];
-    bufF=new float[KINECT_TAGS_DEPTH_WIDTH*KINECT_TAGS_DEPTH_HEIGHT];
-    bufFPl=new float[KINECT_TAGS_DEPTH_WIDTH*KINECT_TAGS_DEPTH_HEIGHT];
+    buf=new unsigned short[depth_width*depth_height];
```
Comment (pattacini) [inline library/src/kinectWrapper_server.cpp]: @Tobias-Fischer where are the corresponding `delete's`?

File: library/src/kinectDriverOpenNI.cpp [hunk 2]
```diff
@@ -89,29 +89,29 @@ bool KinectDriverOpenNI::initialize(Property &opt)
     this->seatedMode=false;
     this->img_width=opt.check("img_width",Value(320)).asInt();
     this->img_height=opt.check("img_height",Value(240)).asInt();
+    this->depth_width=opt.check("depth_width",Value(320)).asInt();
```
Comment (pattacini) [inline library/src/kinectDriverOpenNI.cpp]: the newly introduced options `depth_width`, `depth_height` and `device` must be documented in the interface (I guess in the `kinectWrapper::open` method).

File: library/src/kinectWrapper_server.cpp (hunk 1 above)
Comment (Tobias-Fischer) [inline library/src/kinectWrapper_server.cpp]: Hi @pattacini,
in the server, the `delete`'s are found in `threadRelease`. However, I forgot to free the memory in the client, which is addressed in https://github.com/Tobias-Fischer/kinect-wrapper/commit/eb791a30d8d73e5b20b33c1e1fae1fba53300b4a

Thanks for pointing that out!

File: library/src/kinectDriverOpenNI.cpp (hunk 2 above)
Comment (Tobias-Fischer) [inline library/src/kinectDriverOpenNI.cpp]: Hi, the options were already documented in `modules/kinectServer/src/main.cpp`. To be consistent, I also added them in ` library/include/kinectWrapper/kinectDriver.h` (and fixed `image_width` -> `img_width`). Respective commit: https://github.com/Tobias-Fischer/kinect-wrapper/commit/c2dba62f917b56a2d19c1a58344c92c137170d2c

**Comment by Tobias-Fischer** [comment]
Thanks for merging @pattacini!

**Pull request merged**